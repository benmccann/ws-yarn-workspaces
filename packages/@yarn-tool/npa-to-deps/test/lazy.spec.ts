import { basename, extname } from 'path';
import { npaToDepsValue } from '../index';

describe(`should return semver`, () =>
{

	[
		'botkit@jonchurch/botkit#multi-hears',
		'botkit@12',
	].forEach((input) =>
	{

		test(input, () =>
		{

			let actual = npaToDepsValue(input);

			expect(actual.semver.length).toBeGreaterThanOrEqual(1);

			expect(actual).toMatchSnapshot();

		});

	});

});

describe(`should return undefined`, () =>
{

	[
		'botkit',
		'botkit@^12',
	].forEach((input) =>
	{

		test(input, () =>
		{

			let actual = npaToDepsValue(input);

			expect(actual).toMatchSnapshot({
				fetchQuery: true,
			});

		});

	});

});
