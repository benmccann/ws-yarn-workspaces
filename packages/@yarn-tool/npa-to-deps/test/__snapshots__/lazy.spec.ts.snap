// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should return semver botkit@12 1`] = `
Object {
  "fetchQuery": true,
  "name": "botkit",
  "operator": undefined,
  "result": Object {
    "escapedName": "botkit",
    "fetchSpec": "12",
    "gitCommittish": undefined,
    "gitRange": undefined,
    "name": "botkit",
    "raw": "botkit@12",
    "rawSpec": "12",
    "registry": true,
    "saveSpec": null,
    "scope": undefined,
    "type": "range",
    "where": undefined,
  },
  "semver": "12",
}
`;

exports[`should return semver botkit@jonchurch/botkit#multi-hears 1`] = `
Object {
  "fetchQuery": undefined,
  "name": "botkit",
  "operator": undefined,
  "result": Object {
    "escapedName": "botkit",
    "fetchSpec": null,
    "gitCommittish": "multi-hears",
    "gitRange": undefined,
    "name": "botkit",
    "raw": "botkit@jonchurch/botkit#multi-hears",
    "rawSpec": "jonchurch/botkit#multi-hears",
    "registry": undefined,
    "saveSpec": "github:jonchurch/botkit#multi-hears",
    "scope": undefined,
    "type": "git",
    "where": undefined,
  },
  "semver": "github:jonchurch/botkit#multi-hears",
}
`;

exports[`should return undefined botkit 1`] = `
Object {
  "fetchQuery": true,
  "name": "botkit",
  "operator": "^",
  "result": Object {
    "escapedName": "botkit",
    "fetchSpec": "latest",
    "gitCommittish": undefined,
    "gitRange": undefined,
    "name": "botkit",
    "raw": "botkit",
    "rawSpec": "",
    "registry": true,
    "saveSpec": null,
    "scope": undefined,
    "type": "tag",
    "where": undefined,
  },
  "semver": undefined,
}
`;

exports[`should return undefined botkit@^12 1`] = `
Object {
  "fetchQuery": true,
  "name": "botkit",
  "operator": "^",
  "result": Object {
    "escapedName": "botkit",
    "fetchSpec": "^12",
    "gitCommittish": undefined,
    "gitRange": undefined,
    "name": "botkit",
    "raw": "botkit@^12",
    "rawSpec": "^12",
    "registry": true,
    "saveSpec": null,
    "scope": undefined,
    "type": "range",
    "where": undefined,
  },
  "semver": undefined,
}
`;
